modules/app.xql

```diff
@@ -72,6 +69,143 @@ declare function local:report-table($reports as node()*, $param as xs:string?) a
     </table>
 };
 
+(:
+    Creates the breadcrumb menu for all pages
+:)
+declare function app:breadcrumb($node as node(), $model as map(*)) as element()?{
+   if (ends-with(request:get-uri(),'index.html'))
+   then ()
+   else
+    let $crumbs := reverse(local:breadcrumb($node, $model))
+    return
+        <nav aria-label="breadcrumb" class="col-md-12">
+            <ol class="breadcrumb">
+                <li class="breadcrumb-item"><a href="index.html">Home</a></li>
+                {for $crumb in $crumbs return 
+                    if (functx:index-of-node($crumbs, $crumb) lt count($crumbs))
+                    then
+                    <li class="breadcrumb-item">{$crumb}</li>
+                    else
+                    <li class="breadcrumb-item active" aria-current="page">{string($crumb)}</li>
+                  }
+             </ol>
+         </nav>
+};
+
+(:
+    Returns the breadcrumb path for a given document. This function determines
+    what kind of page we're on and then calls the function associated with that
+    page type.
+:)
+declare function local:breadcrumb($node as node(), $model as map(*)) as item()*{
+let $uri := tokenize(request:get-uri(),'/')[last()]
+let $id := (request:get-parameter('f',request:get-parameter('a',())))
+let $doc := if ($id) then collection:fetch($id) else ()
```

Comment:
Would it be better to use `$model('document')` instead of `collection:fetch($id)` here? The document should already be loaded into the model, so we shouldn't need to load it twice.